### PR TITLE
Reworks the security module for cyborgs to be more useful.

### PR DESCRIPTION
**Pull request by TheRyeGuyWhoWillNowDie**
# Document the changes in your pull request

Reworks the security cyborg! As a whole they're now stronger, although their ranged capabilities have been limited. They're also called Controller cyborgs now because of their extensive refitting.

Their new equipment:
 
- Flash
- Zip Ties
- Stun baton
- Cyborg TRIPnet launcher: A self-charging DRAGnet that does more stamina damage on netting mode (16 instead of 10), at the cost of being unable to teleport.
- Tear gas grenade launcher: Can hold up to 3 tear gas grenades. Refills itself in a cyborg recharger.
- Cyborg holobarrier projector: Projects up to 4 reinforced holobarriers, impassible on walk intent and have more health, but take longer to project and can project less.
- Cyborg verbal dissuasion tool: Identical to their previous hailer, but can no longer request backup.
- When hacked/emagged, they gain a gun that fires DRAGnet netting, and can teleport.

**They're now flash resistant too, functionally identical to the syndicate assault cyborg.**
However, they're slower now, moving at the same speed as someone in a space suit to prevent them from chasing people and spamming TRIPnet shots. They're here to keep the peace, not hunt criminals.
Silicons can also no longer interact with brig timers, they shouldn't be arresting people anyways, and they can still open the door if need be.
Controller cyborgs also don't have the security encryption key anymore, they're not part of security.

The reasoning:
Currently, security modules are in an odd place. In the past; they were extremely strong and borderline forced antagonists to make/buy EMPs to avoid getting rushed by a VTEC borg with a taser, but this hasn't been the case for years. Currently, security cyborgs are mostly only played on malf AI for the disabler and lasergun, because of how useless they are normally. Their disabler only has 10 shots, they have damage slowdown, and their baton cannot be upgraded. All while retaining the cyborg vulnerabilities of EMPs and flashes being an instant game over. By reworking them into controller cyborgs, they're much better at guarding areas, which cyborgs are already good at due to their ability to bolt doors on command. Their potential to hunt for valids has been reduced by limiting their ranged capabilities and slowing them, because security cyborgs searching for valids with AA has always been despised. I made this rework in the hopes to make security cyborg played more often, and generally more fun for both sides of the baton, so to speak. They're still a module that requires a module picker (still locked behind advanced weapons tech) because of their increased capabilities. 

# Spriting
Current sprite for the module, stolen from [this /tg/ pull request.](https://github.com/tgstation/tgstation/pull/66074)
image
image
image
image

# Wiki Documentation

Cyborg modules page, Guide to Combat

# Changelog

:cl:  
rscadd: Adds the controller cyborg module
rscdel: Removes the security cyborg module 
tweak: cyborg hailer has been renamed and can no longer call for backup
tweak: silicons can no longer interact with brig timers  
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/brigdoors.dm [hunk 1]
```diff
@@ -210,10 +210,14 @@
 	timer_duration = new_time
 
 /obj/machinery/door_timer/ui_interact(mob/user, datum/tgui/ui)
-	ui = SStgui.try_update_ui(user, src, ui)
-	if(!ui)
-		ui = new(user, src, "BrigTimer", name)
-		ui.open()
+	if (!issilicon(user))
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm]: early return

File: code/game/objects/items/holosign_creator.dm [hunk 2]
```diff
@@ -74,6 +74,13 @@
 	creation_time = 30
 	max_signs = 6
 
+/obj/item/holosign_creator/security/cyborg
+	name = "cyborg holobarrier projector"
+	desc = "Projects reinforced holobarriers that are impassable and tougher, at the cost of taking longer to project."
+	holosign_type = /obj/structure/holosign/barrier/reinforced
+	creation_time = 50
```
Comment (Bibby0110) [inline code/game/objects/items/holosign_creator.dm]:
```suggestion
	creation_time = 5 SECONDS
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -899,8 +899,8 @@
 	new_module = /obj/item/robot_module/clown
 
 /obj/item/borg/upgrade/transform/security
```
Comment (Bibby0110) [inline code/game/objects/items/robot/robot_upgrades.dm]: rename the typepath if you're so insistent on this not being a security borg

File: code/game/objects/structures/holosign.dm [hunk 4]
```diff
@@ -56,6 +56,12 @@
 	max_integrity = 20
 	var/allow_walk = 1 //can we pass through it on walk intent
 
+/obj/structure/holosign/barrier/reinforced
+	name = "durable holobarrier"
+	desc = "A durable holobarrier. Harder to destroy and cannot be passed through by walking."
+	max_integrity = 30
+	allow_walk = 0
```
Comment (Bibby0110) [inline code/game/objects/structures/holosign.dm]:
```suggestion
	allow_walk = FALSE
```

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 5]
```diff
@@ -879,6 +879,7 @@
 /mob/living/silicon/robot/modules/security
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/robot/robot.dm]: ditto on typepath change

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 6]
```diff
@@ -359,18 +362,16 @@
 	hat_offset = -4
 
 /obj/item/robot_module/security
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: ditto

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 7]
```diff
@@ -418,7 +419,7 @@
 /obj/item/robot_module/peacekeeper/do_transform_animation()
 	..()
 	to_chat(loc, "<span class='userdanger'>Under ASIMOV, you are an enforcer of the PEACE and preventer of HUMAN HARM. \
-	You are not a security module and you are expected to follow orders and prevent harm above all else. Space law means nothing to you.</span>")
+	You are not a controller module and you are expected to follow orders and prevent harm above all else. Space law means nothing to you.</span>")
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: wrong place, this is for peacekeeper borgs

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 8]
```diff
@@ -71,43 +71,45 @@
 	ammo_x_offset = 1
 	weapon_weight = WEAPON_MEDIUM
 	var/obj/item/beacon/teletarget = null
+	var/cantelly = 1
```
Comment (Bibby0110) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: use a proper variable name, also TRUE/FALSE

File: code/modules/projectiles/projectile/energy/net_snare.dm [hunk 9]
```diff
@@ -6,22 +6,29 @@
 	hitsound = 'sound/weapons/taserhit.ogg'
 	range = 10
 	var/obj/item/beacon/teletarget = null
+	var/cantelly = 1
```
Comment (Bibby0110) [inline code/modules/projectiles/projectile/energy/net_snare.dm]: ditto as above

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 10]
```diff
@@ -71,43 +71,45 @@
...
-		user.show_message(span_notice("Random teleport enabled."))
-	else
-		var/obj/item/beacon/T = L[t1]
-		var/area/A = get_area(T)
-		if(A.noteleport)
-			to_chat(user, span_notice("\The [src] is malfunctioning."))
-			return
-		current_location = get_turf(user)	//Recheck.
-		current_area = current_location.loc
+	if(cantelly)
```
Comment (Bibby0110) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: early return

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 7 above)
Comment (TheRyeGuyWhoWillNowDie) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: changed for the controller transform text also, this is intentional

**Comment by maxion12345** [comment]
on one hand, there are times when Borg's shouldn't valid, and this is a good loadout. However, there are also times where borgs need to go fuck shit up (see: gangs, revs, wizards, lings, shadowlings, darkspawn, etc etc)

Also brig timers are a machine, silicon's interact with machines. At most this will make a borg just perma people because they don't need to follow space law. There are also lawsets that make the AI interact with space law (robocop for instance) and the AI would need these interactions to act out their laws.

If it had more combat capabilities it might be an interesting side grade but overall it's a pretty big nerf to security borgs. The flash suppression is a nice idea but that just means EMPs will become even more the meta. The idea of riot suppression gear is neat though, could make it an upgrade for security borgs that'll give me the holoemitter and tear gas launchers

**Comment by TheRyeGuyWhoWillNowDie** [comment]
> on one hand, there are times when Borg's shouldn't valid, and this is a good loadout. However, there are also times where borgs need to go fuck shit up (see: gangs, revs, wizards, lings, shadowlings, darkspawn, etc etc)
> 
> Also brig timers are a machine, silicon's interact with machines. At most this will make a borg just perma people because they don't need to follow space law. There are also lawsets that make the AI interact with space law (robocop for instance) and the AI would need these interactions to act out their laws.
> 
> If it had more combat capabilities it might be an interesting side grade but overall it's a pretty big nerf to security borgs. The flash suppression is a nice idea but that just means EMPs will become even more the meta. The idea of riot suppression gear is neat though, could make it an upgrade for security borgs that'll give me the holoemitter and tear gas launchers

I'm fairly confident that they'll perform better in most situations. The TRIPnet fires 6 shots that do 16 stamina each, and you're bound to hit at least 2 (32 stamina per shot) instead, which outclasses the disabler's 30 stamina per shot. It's not like they're useless when fighting solo antags like Wizards, since they can setup barricades and snare traps and go in with the TRIPnet. I'd argue they're more effective against group antagonists like Revolutionaries and Gangs because of the tear gas grenades and spread-out TRIPnet shots, and because of the fact that the Revolutionaries' flash isn't an instant game-over tool against them. The brig timer concern is valid, and I am willing to change it or give Controller borgs a warning that they're not security when they try to use it but let them use it normally. I'm also willing to reduce the slowdown on them a little if it's a big concern as well.

**Comment by JamieD1** [comment]
Your argument about them not being sec, but you give them Tear gas and flash protection and a Dragnet?
You dont see them that often because they are locked behind tech and not round start. If they were not you would see alot more of them, Im sorry but this really doesnt feel like a good idea to make them not even have sec comms either. Like why do they even exist. Borgs are allowed to help security if it is within their lawset to do so.

You would be better off giving a tool or two to the peacekeeping borg. Your attempt to replace a sec borg with a stronger sec borg i'm sorry is not what we want.

**Pull request closed without merging**